lib/flip_transition.dart

```diff
@@ -0,0 +1,182 @@
+import 'dart:math';
+
+import 'package:flip_card/flip_card.dart';
+import 'package:flutter/material.dart';
+
+Widget _fill(Widget child) => Positioned.fill(child: child);
+Widget _noop(Widget child) => child;
+
+class FlipTransition extends StatefulWidget {
+  const FlipTransition({
+    super.key,
+    required this.front,
+    required this.back,
+    required this.animation,
+    this.direction = Axis.horizontal,
+    this.fill = Fill.none,
+    this.alignment = Alignment.center,
+    this.frontAnimator,
+    this.backAnimator,
+  });
+
+  final Widget front;
+  final Widget back;
+  final Animation<double> animation;
+  final Axis direction;
+  final Fill fill;
+  final Alignment alignment;
+
+  final Animatable<double>? frontAnimator;
+  final Animatable<double>? backAnimator;
+
+  static final defaultFrontAnimator = TweenSequence(
+    [
+      TweenSequenceItem<double>(
+        tween: Tween(begin: 0.0, end: pi / 2).chain(
+          CurveTween(curve: Curves.easeIn),
+        ),
+        weight: 50.0,
+      ),
+      TweenSequenceItem<double>(
+        tween: ConstantTween<double>(pi / 2),
+        weight: 50.0,
+      ),
+    ],
+  );
+
+  static final defaultBackAnimator = TweenSequence(
+    [
+      TweenSequenceItem<double>(
+        tween: ConstantTween<double>(pi / 2),
+        weight: 50.0,
+      ),
+      TweenSequenceItem<double>(
+        tween: Tween(begin: -pi / 2, end: 0.0).chain(
+          CurveTween(curve: Curves.easeOut),
+        ),
+        weight: 50.0,
+      ),
+    ],
+  );
+
+  @override
+  State<FlipTransition> createState() => _FlipTransitionState();
+}
+
+class _FlipTransitionState extends State<FlipTransition> {
+  late CardSide _currentSide;
+
+  @override
+  void initState() {
+    super.initState();
+    _currentSide = _getSideFor(widget.animation.status);
+    widget.animation.addStatusListener(_handleChange);
+  }
+
+  @override
+  void didUpdateWidget(covariant FlipTransition oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (widget.animation != oldWidget.animation) {
+      oldWidget.animation.removeStatusListener(_handleChange);
+      widget.animation.addStatusListener(_handleChange);
+      _currentSide = _getSideFor(widget.animation.status);
+    }
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+    widget.animation.removeStatusListener(_handleChange);
+  }
+
+  CardSide _getSideFor(AnimationStatus status) {
+    switch (status) {
```

Comment:
This seems dangerous, example:
1. user taps to flip
2. flip animation starts (`AnimationStatus.forward`)
3. app uses the _currentSide property (returns `CardSide.back` even though the front of the card is still visible)

Real world impact is probably relatively small, but I'd consider this a bug or at least a breaking change since the previous version of this package only updated the "active" card state once the animation completed (if I recall correctly!).

Reply:
If the user taps the card before the animation is completed, the animation will be reversed to the other side.
In this case when animating towards the back side of the card, it will start animating towards front again.

Reply:
Got it, makes sense